### DISCUSSION TITLE
v1.11.0

### DISCUSSION THREAD

**Discussion by satabin** [satabin/fs2-data#580] [Announcements]
This release brings new feature in the pretty printing domain!

 - **Project**
   - Update copyright and clarify it in CONTRIBUTING guidelines (#564 by @satabin)
 - **Common**
   - Add streaming pretty printing infrastructure (#567 by @satabin)
 - **JSON**
   - Use the new pretty printer under the hood  (#567 by @satabin)
 - **XML**
   - Add pretty printing for XML  (#565 and #567 by @satabin)

Following dependency updates are included (by @scala-steward):
 - Update sbt-scalajs, scalajs-compiler, ... to 1.16.0
 - Update fs2-core, fs2-io to 3.10.0
 - Update sbt-assembly to 2.2.0
 - Update scala3-library, ... to 3.3.3
 - Update sbt-typelevel, sbt-typelevel-site to 0.6.7
 - Update scala-library, scala-reflect to 2.12.19
 - Update kind-projector to 0.13.3
 - Update sbt to 1.9.9
 - Update scalafmt-core to 3.8.0
 - Update weaver-cats, weaver-scalacheck to 0.8.4
 - Update nscplugin, sbt-scala-native, ... to 0.4.17
 - Update fs2-core to 3.9.4
 - Update play-json to 3.0.2
 - Update shapeless3-deriving to 3.4.1
 - Update sbt-jmh to 0.4.7

For a complete list of issues and PRs, see [the milestone](https://github.com/gnieh/fs2-data/milestone/27?closed=1)

This release was brought to you by the following contributors:
```
git shortlog -sn --no-merges v1.10.0..v1.11.0
    23  Lucas Satabin
    20  Scala Steward
```

<hr /><em>This discussion was created from the release <a href='https://github.com/gnieh/fs2-data/releases/tag/v1.11.0'>v1.11.0</a>.</em>

**Comment by lenguyenthanh** [comment]
Hi, I think this update changes the semantic of `codec.serialize` or `render.compact`(I believe the latter, but I haven't checked the related pr 🤷). The reason is it broke my anti regression tests for: https://github.com/lichess-org/lila-fishnet/pull/292/files. The fix is change `lastOrError` to `foldMonoid`. But I'm not sure that's the right thing to do.

Edit: It doesn't break my code though, only test.

**Reply by satabin** [reply]
Hi

Thanks for the feedback. It might indeed change chunking in some cases. It happened to be in one chunk before and now in several ones. If you expect to have the whole result a string, you might want to use the `.compile.string` instead. It will also probably be more performant than `foldMonoid` in this case.

**Comment by lenguyenthanh** [comment]
That makes sense, thanks! I should haven't relied on the mechanism of chunking at the first place.

btw, thanks for the library, I really love it and use it a lot!

**Reply by satabin** [reply]
I am really glad to read it. Would you mind if I put lichess as an adopter on the website?

**Reply by lenguyenthanh** [reply]
No, I definitely don't. But I'm asking the team just to make sure. I'll create a PR when I have their mercy.

**Reply by lenguyenthanh** [reply]
done: https://github.com/gnieh/fs2-data/pull/583
